### ISSUE TITLE
Strange song skipping bugs & music starting after LMS restart since 7.9.2

### ISSUE THREAD

**Issue by tstabrawa** [LMS-Community/slimserver#532]
About 1-2 years ago, I noticed that after upgrading LMS, that I started seeing strange problems with tracks skipping as synced players come back online.  For the most part, I've just lived with it, but finally had the chance to dig into it further.

The problems can be summed up by the following scenarios:

1. Multiple players synced, playing Pandora.  One of the synced players has been offline for several hours, then comes back online (e.g. SqueezePlayer Android app is started with the intention of listening to the current song in another room).  When the additional player comes online, Pandora skips to the next song.
2. Multiple players synced, playing a normal playlist of local songs.  One of the synced players has been offline for several hours, then comes back online.  When the additional player comes online, the playlist advances to the last song and stops playing on all synced players.
3. Multiple players synced, all paused.  LMS is restarted (e.g. server is restarted by unattended-updates or by Proxmox taking a backup of the container/virtual machine).  Upon LMS resuming, all synced players start playing.  Often the music that starts is not the same as what was paused before the restart (e.g. a different Pandora station may be started).  I've had multiple nights where my wife said she had to go downstairs to turn off "Machine Head Radio" playing throughout the house in the middle of the night.

If memory serves me correctly, I did not experience any of these problems on version 7.9.1, but I did see them after installing a nightly pre-release version of 7.9.2 (specifically, logitechmediaserver_7.9.2\~1571226194_amd64.deb).  I also can confirm that the problems still exist as of a recent 8.2.0 nightly version (logitechmediaserver_8.2.0~1612378864_amd64.deb).

As far as I can tell, these problems were introduced by PR #249.  Specifically, if I revert commits 6d1694c0f2105a6cbfd596b7dbf04c4508c2b258 and a048a8fb7f391c36dda4425e31489fe393f64253 (by manually editing the files in an installed copy of logitechmediaserver_8.2.0~1612378864_amd64.deb), then all three problems go away entirely.  Unfortunately, I expect that reverting these commits would completely break the feature that was added in PR #249 (re-starting playback of non-synced players when they come online after being unplugged for a while).  I tried digging around in the code to see if I could identify a way to preserve that feature without causing the three problems listed above, but I just don't have a good enough familiarity with this codebase yet to propose a specific solution that wouldn't break something else.  Hopefully having identified the symptoms and the commits, someone with more experience on this project might be able to help, though.

Please let me know if you need additional information or if you'd like me to test any patches.

**Comment by tstabrawa** [comment]
The following is the change that I applied after installing logitechmediaserver_8.2.0~1612378864_amd64.deb which caused the problems to go away.  (I wouldn't necessarily suggest actually making this specific change in a release, because it would probably break #249.)

```
/usr/share/perl5/Slim/Player$ diff -c Player.pm.lfc Player.pm
*** Player.pm.lfc       2009-08-03 10:14:33.000000000 -0500
--- Player.pm   2021-02-06 23:59:46.190225805 -0600
***************
*** 109,116 ****
        Slim::Buttons::Home::updateMenu($client);

        # fire it up!
        $client->startup($syncgroupid);
-       $client->power($prefs->client($client)->get('power'), 0, 1);

        return if $client->display->isa('Slim::Display::NoDisplay');

--- 109,116 ----
        Slim::Buttons::Home::updateMenu($client);

        # fire it up!
+       $client->power($prefs->client($client)->get('power'));
        $client->startup($syncgroupid);

        return if $client->display->isa('Slim::Display::NoDisplay');
```

**Comment by tstabrawa** [comment]
Upon updating to logitechmediaserver_8.4.0~1670141288_amd64.deb, I can confirm that all three scenarios described above are working as expected now.  As far as I can tell, this was fixed by PR #699, which reverted PR #249 (among other things).

Thanks @philippe44!  Closing this issue.
